packages/backend-core/src/middleware/authenticated.ts

```diff
@@ -48,22 +48,31 @@ async function checkApiKey(apiKey: string, populateUser?: Function) {
   const decrypted = decrypt(apiKey)
   const tenantId = decrypted.split(SEPARATOR)[0]
   return doInTenant(tenantId, async () => {
-    const db = getGlobalDB()
-    // api key is encrypted in the database
-    const userId = (await queryGlobalView(
-      ViewName.BY_API_KEY,
-      {
-        key: apiKey,
-      },
-      db
-    )) as string
+    let userId
+    try {
+      const db = getGlobalDB()
+      // api key is encrypted in the database
+      userId = (await queryGlobalView(
+        ViewName.BY_API_KEY,
+        {
+          key: apiKey,
+        },
+        db
+      )) as string
+    } catch (err) {
+      userId = undefined
+    }
     if (userId) {
       return {
         valid: true,
         user: await getUser(userId, tenantId, populateUser),
       }
     } else {
-      throw "Invalid API key"
+      throw {
+        message:
+          "Invalid API key - may need re-generated, or user doesn't exist",
+        name: "InvalidApiKey",
```

Comment:
I think this would be a great candidate to build out the errors framework. I was gonna suggest how to do this, but realised that the existing framework was unnecessarily complex and hard to understand around `type`, `code` and `context`. It wasn't that easy to extend. 

I've updated it here: https://github.com/Budibase/budibase/pull/10042

This updat makes it much easier to just drop in a new section:
```ts
// AUTH

export class InvalidAPIKey extends BudibaseError {

  constructor(message: string) {
    super(message, ErrorCode.INVALID_API_KEY)
    this.status = httpStatus
  }
}
```

The catch block can then be updated to
```ts
} else if (err?.code ===  ErrorCode.INVALID_API_KEY) {
```